drivers/hmis/lib/hmis_util/service_types.rb

```diff
@@ -141,24 +121,22 @@ def self.seed_hud_service_types(data_source_id)
     # Ensures that Form Instances exist for each HUD Service Type according to the configuration.
     def self.seed_hud_service_form_instances
       HUD_SERVICE_INSTANCE_CONFIG.each do |config|
-        (record_type, type_provided, project_types) = config.values_at(:record_type, :type_provided, :project_types)
-
-        csts = Hmis::Hud::CustomServiceType.where(hud_record_type: record_type)
+        (record_type, project_types, funders) = config.values_at(:record_type, :project_types, :funders)
 
-        custom_service_type = csts.where(hud_type_provided: type_provided).first if type_provided.present?
-        custom_service_category = csts.first.custom_service_category unless type_provided.present?
-        next unless custom_service_type.present? || custom_service_category.present?
+        custom_service_category = Hmis::Hud::CustomServiceType.where(hud_record_type: record_type).first&.custom_service_category
+        raise "Category not found for record type #{record_type}" unless custom_service_category.present?
 
-        # puts "Ensuring '#{custom_service_category&.name}' instances for #{project_types.size} project types"
+        # Create a Form Instance for each combination of project type + funder
         (project_types || [nil]).each do |project_type|
-          Hmis::Form::Instance.where(
-            entity_type: 'ProjectType',
-            entity_id: project_type,
-            definition_identifier: 'service', # The default definition used for all HUD Services
-            custom_service_type_id: custom_service_type&.id,
-            custom_service_category_id: custom_service_category&.id,
-            # funder: funder,
-          ).first_or_create
+          (funders || [nil]).each do |funder|
+            puts "Creating instance for '#{custom_service_category&.name}' ProjectType:#{project_type} Funder:#{funder}"
+            Hmis::Form::Instance.where(
+              definition_identifier: 'service', # The default definition used for all HUD Services
+              custom_service_category_id: custom_service_category&.id,
```

Comment:
small note, since we raise on ln 127, I think we don't need safe safe navigation (`&.`) since custom_service_category should always be present

Reply:
Ah true, that was left over from a previous version where that wasn't the case